go/kbfs/libkbfs/kbfs_cr_test.go

```diff
@@ -2025,3 +2025,95 @@ func TestForceStuckConflict(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, children, 1)
 }
+
+// Tests that if clearing a CR conflict can fast-forward if needed.
+func TestBasicCRFailureClearAndFastForward(t *testing.T) {
+	// simulate two users
+	var userName1, userName2 kbname.NormalizedUsername = "u1", "u2"
+	config1, _, ctx, cancel := kbfsOpsConcurInit(t, userName1, userName2)
+	defer kbfsConcurTestShutdown(ctx, t, config1, cancel)
+
+	config2 := ConfigAsUser(config1, userName2)
+	defer CheckConfigAndShutdown(ctx, t, config2)
+
+	// Enable journaling on user 2
+	tempdir, err := ioutil.TempDir(os.TempDir(), "journal_for_fail_fix")
+	defer os.RemoveAll(tempdir)
+	require.NoError(t, err)
+	err = config2.EnableDiskLimiter(tempdir)
+	require.NoError(t, err)
+	err = config2.EnableJournaling(ctx, tempdir,
+		TLFJournalBackgroundWorkEnabled)
+	require.NoError(t, err)
+	jManager, err := GetJournalManager(config2)
+	require.NoError(t, err)
+	err = jManager.EnableAuto(ctx)
+	require.NoError(t, err)
+
+	name := userName1.String() + "," + userName2.String()
+
+	t.Log("User 1 creates a file a/b.")
+	rootNode1 := GetRootNodeOrBust(ctx, t, config1, name, tlf.Private)
+
+	kbfsOps1 := config1.KBFSOps()
+	dirA1, _, err := kbfsOps1.CreateDir(ctx, rootNode1, testPPS("a"))
+	require.NoError(t, err)
+	fileB1, _, err := kbfsOps1.CreateFile(
+		ctx, dirA1, testPPS("b"), false, NoExcl)
+	require.NoError(t, err)
+	err = kbfsOps1.SyncAll(ctx, rootNode1.GetFolderBranch())
+	require.NoError(t, err)
+
+	t.Log("User 2 looks up the file node.")
+	rootNode2 := GetRootNodeOrBust(ctx, t, config2, name, tlf.Private)
+	kbfsOps2 := config2.KBFSOps()
+	dirA2, _, err := kbfsOps2.Lookup(ctx, rootNode2, testPPS("a"))
+	require.NoError(t, err)
+	fileB2, _, err := kbfsOps2.Lookup(ctx, dirA2, testPPS("b"))
+	require.NoError(t, err)
+
+	t.Log("Force a conflict")
+	tlfID := rootNode2.GetFolderBranch().Tlf
+	err = kbfsOps2.ForceStuckConflictForTesting(ctx, tlfID)
+	require.NoError(t, err)
+
+	t.Log("User 1 updates mod time on a/b.")
+
+	mtime := time.Now()
+	for i := 0; i < fastForwardRevThresh+2; i++ {
+		mtime = mtime.Add(1 * time.Minute)
+		err = kbfsOps1.SetMtime(ctx, fileB1, &mtime)
+		err = kbfsOps1.SyncAll(ctx, fileB1.GetFolderBranch())
+		require.NoError(t, err)
+	}
+
+	t.Log("Ensure only conflict files are in the conflict view")
+	children, err := kbfsOps2.GetDirChildren(ctx, rootNode2)
+	require.NoError(t, err)
+	require.Len(t, children, 1+(maxConflictResolutionAttempts+1))
```

Comment:
Why is this +2? (same q for below too)

Reply:
The `ForceStuckConflictForTesting` call makes `maxConflictResolutionAttempts+1` new files in the root directory while it is forcing the conflict, and the directory `a` was made earlier by the test.